registry/handlers/hooks.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"fmt"
+	"net/smtp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/configuration"
+)
+
+// ReportHook is for hooking Panic in web application
+type ReportHook struct {
```

Comment:
This should be called something like `EmailHook` and include the smtp configuration.